src/main/java/fi/digitraffic/tis/vaco/crypt/EncryptionService.java

```diff
@@ -75,4 +96,81 @@ public <T> T decrypt(String cypher, Class<T> clz) {
             throw new VacoException("Failed to deserialize decrypted result", e);
         }
     }
+
+    @Value("${vaco.encryptionKeys.credentials}")
+    private String keyId;
+
+    private static KmsAsyncClient kmsAsyncClient;
+
+    public byte[] encryptBlob(AuthenticationDetails details) {
+
+            SdkBytes myBytes = SdkBytes.fromUtf8String(details.toString());
+            EncryptRequest encryptRequest = EncryptRequest.builder()
+                .keyId(keyId)
+                .plaintext(myBytes)
+                .build();
+
+            try {
+                EncryptResponse response = getAsyncClient()
+                    .encrypt(encryptRequest)
+                    .toCompletableFuture()
+                    .get();
+
+                return response.ciphertextBlob().asByteArray();
+            } catch (InterruptedException | ExecutionException e) {
+                throw new RuntimeException("Error during encryption", e);
+            }
+    }
+
+    public byte[] decryptBlob(byte[] encryptedData) {
+
+        SdkBytes bytes = SdkBytes.fromByteArray(encryptedData);
+        logger.debug("Decrypting data: {}", Base64.getEncoder().encodeToString(encryptedData));
```

Comment:
Our log digester system probably doesn't like this as this'll be byte mush, plus the array might be quite big so thge log row may end up being huge. Maybe replace with length of the byte array or something less noisy and confusing.